### PR TITLE
Refactor costing

### PR DESCRIPTION
**Pull request by iamyulong**
## Summary

Some refactors to costing in preparation for further updates.

## Details

- Moved all fee configuration and rules (except instrumentation parameters) to `FeeTable`
- Combined all costing entry to `CostingEntry`
- Removed costing from native blueprints
- Moved costing trace to `CostingModule` 
- Added script to scan all native functions and detect misconfiguration
- Added gas buffer to Scrypto Runtime

## Limitations

**The costing numbers are completely off after this PR!**

Will update them in the following weeks.

### CODE REVIEW

File: radix-engine/src/system/system_modules/limits/module.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@ use crate::{
 };
 
 #[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
-pub enum TransactionLimitsError {
+pub enum LimitingError {
```
Comment (talekhinezh) [inline radix-engine/src/system/system_modules/limits/module.rs:17]: Weird name? What is wrong with the original name?

File: radix-engine/src/vm/wasm/constants.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use radix_engine_constants::DEFAULT_MAX_WASM_MEM_PER_CALL_FRAME;
 
 pub const CONSUME_BUFFER_FUNCTION_NAME: &str = "consume_buffer";
-pub const CONSUME_COST_UNITS_FUNCTION_NAME: &str = "gas";
+pub const CONSUME_GAS_FUNCTION_NAME: &str = "gas";
```
Comment (talekhinezh) [inline radix-engine/src/vm/wasm/constants.rs:4]: Gas name is back?

File: radix-engine/src/vm/wasm/traits.rs [hunk 3]
```diff
@@ -127,7 +127,7 @@ pub trait WasmRuntime {
 
     fn assert_access_rule(&mut self, rule: Vec<u8>) -> Result<(), InvokeError<WasmRuntimeError>>;
 
-    fn consume_cost_units(&mut self, n: u32) -> Result<(), InvokeError<WasmRuntimeError>>;
+    fn consume_gas(&mut self, n: u32) -> Result<(), InvokeError<WasmRuntimeError>>;
```
Comment (talekhinezh) [inline radix-engine/src/vm/wasm/traits.rs:130]: gas?

**Review by talekhinezh — APPROVED** [review]

File: radix-engine/src/system/system_modules/limits/module.rs (hunk 1 above)
Comment (iamyulong) [reply radix-engine/src/system/system_modules/limits/module.rs:17]: Wanted to be consistent with `CostingError`, but happy to change.

File: radix-engine/src/vm/wasm/constants.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/src/vm/wasm/constants.rs:4]: I've added to "To discuss" list.

File: radix-engine/src/vm/wasm/traits.rs (hunk 3 above)
Comment (iamyulong) [reply radix-engine/src/vm/wasm/traits.rs:130]: Same.

File: radix-engine/src/vm/wasm/constants.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/src/vm/wasm/constants.rs:4]: Reverted!

**Pull request merged**